### PR TITLE
Fixed LogitsProcessor

### PR DESCRIPTION
**Pull request by jkrukowski**
Issue is described [here](https://github.com/huggingface/swift-transformers/issues/44)

When `LogitsProcessor` contains more than one instance of `LogitsWarper` wrong indexes are returned -- we're returning indexes of the processed array but we should return these indexes mapped to the result of the previous call of `LogitsProcessor`.

Fixed by changing the interface of `LogitsWarper` allowing to pass indexes along with logits and mapping the result indexes to the passed indexes.

### CODE REVIEW

File: Tests/TensorUtilsTests/LogitsWarperTests.swift [hunk 1]
```diff
@@ -95,7 +109,14 @@ final class LogitsWarperTests: XCTestCase {
             logitsWarpers: [TopKLogitsWarper(k: 3), TopPLogitsWarper(p: 0.99)]
         )
         let result4 = processor4([2.0, 1.0, 3.0, -5.0, -23.0, 12.5])
-        XCTAssertEqual(result4.indexes, [0])
+        XCTAssertEqual(result4.indexes, [5])
```
Comment (jkrukowski) [inline Tests/TensorUtilsTests/LogitsWarperTests.swift:112]: we had tests for this exact issue but we were asserting the wrong value

File: Sources/TensorUtils/LogitsWarper/TopKLogitsWarper.swift [hunk 2]
```diff
@@ -47,12 +47,12 @@ public struct TopKLogitsWarper: LogitsWarper {
...
-        let distances = bestValues.data!.withMemoryRebound(to: Float.self, capacity: k) { ptr in
+        let topkLogits = bestValues.data!.withMemoryRebound(to: Float.self, capacity: k) { ptr in
             Array(UnsafeBufferPointer(start: ptr, count: k))
         }
-        let indices = bestIndices.data!.withMemoryRebound(to: Int32.self, capacity: k) { ptr in
+        let topkIndexes = bestIndices.data!.withMemoryRebound(to: Int32.self, capacity: k) { ptr in
             Array(UnsafeBufferPointer(start: ptr, count: k))
         }
-        return (indexes: indices.map { Int($0) }, logits: distances)
+        return (indexes: topkIndexes.map { indexes[Int($0)] }, logits: topkLogits)
```
Comment (jkrukowski) [inline Sources/TensorUtils/LogitsWarper/TopKLogitsWarper.swift:56]: here we're mapping indexes

**Review by pcuenca — APPROVED** [review]
Thanks for looking into this so quick! I think we should settle on using the same nomenclature for indexes/indices, but let's fix that bug first.
